astronomer/providers/apache/livy/example_dags/example_livy.py

```diff
@@ -3,38 +3,326 @@
+            },
+        ],
+        "Ec2KeyName": PEM_FILENAME,
+        "KeepJobFlowAliveWhenNoSteps": True,
+        "TerminationProtected": False,
+    },
+    "Steps": SPARK_STEPS,
+    "JobFlowRole": JOB_FLOW_ROLE,
+    "ServiceRole": SERVICE_ROLE,
+}
+
+
+def create_airflow_connection(task_instance: Any) -> None:
+    """
+    Checks if airflow connection exists, if yes then deletes it.
+    Then, create a new livy_default connection.
+    """
+    conn = Connection(
+        conn_id="livy_default",
+        conn_type="livy",
+        host=task_instance.xcom_pull(
+            key="cluster_response_master_public_dns", task_ids=["describe_created_cluster"]
+        )[0],
+        login="",
+        password="",
+        port=8998,
+    )  # create a connection object
+
+    # type should be ignored below as session is by default None and its not Callable.
+    session = settings.Session()
+    connection = session.query(Connection).filter_by(conn_id=conn.conn_id).one_or_none()
+    if connection is None:
+        logging.info("Connection %s doesn't exist.", str(conn.conn_id))
+    else:
+        session.delete(connection)
+        session.commit()
+        logging.info("Connection %s deleted.", str(conn.conn_id))
+
+    session.add(conn)
+    session.commit()  # it will insert the connection object programmatically.
+    logging.info("Connection livy_default is created")
+
+
+def add_inbound_rule_for_security_group(task_instance: Any) -> None:
+    """
+    Sets the inbound rule for the aws security group, based on
+    current ip address of the system.
+    """
+    current_docker_ip = get("https://api.ipify.org").text
+    logging.info("Current ip address is: %s", str(current_docker_ip))
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+
+    response = client.describe_security_groups(
+        GroupIds=[
+            task_instance.xcom_pull(
+                key="cluster_response_master_security_group", task_ids=["describe_created_cluster"]
+            )[0]
+        ]
+    )
+    current_ip_permissions = response["SecurityGroups"][0]["IpPermissions"]
+    ip_exists = False
+    for current_ip in current_ip_permissions:
+        ip_ranges = current_ip["IpRanges"]
+        for ip in ip_ranges:
+            if ip["CidrIp"] == str(current_docker_ip) + "/32":
+                ip_exists = True
+
+    if not ip_exists:
+        # open port for port 8998
+        client.authorize_security_group_ingress(
+            GroupId=task_instance.xcom_pull(
+                key="cluster_response_master_security_group", task_ids=["describe_created_cluster"]
+            )[0],
+            IpPermissions=[
+                {
+                    "IpProtocol": "tcp",
+                    "FromPort": 8998,
+                    "ToPort": 8998,
+                    "IpRanges": [{"CidrIp": str(current_docker_ip) + "/32"}],
+                }
+            ],
+        )
+        logging.info("Added %s/32 for livy with port 8998 open.", str(current_docker_ip))
```

Comment:
I do not think it is a good idea to expose the ip in the logs, can remove it

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/235/commits/67b28cb33bad8c0d314c6f7834d3f52e00cc9177)